serverless/internal/layout/README.md

```diff
@@ -0,0 +1,86 @@
+Layout
+======
+
+This doc describes the on-disk layout and format of the serverless log files.
+
+> :warning: this is still under development, and the exact layout and contents
+> of the serverless log structure may well change.
+
+Broadly, the layout consists of a self-contained directory hierarchy whose
+contents represents the entire state of the log. With one exception, the
+contents and structure of this directory is designed to allow it to be safely
+and indefinitely cached by clients.
+
+Inside the directory you'll find:
+
+ * :file_folder: leaves/
+ * :file_folder: seq/
+ * :file_folder: tile/
+ * :page_facing_up: state
```

Comment:
nit: the order of these not matching the order below is a little distracting.

Reply:
rejiggered